lib/src/value/calculation.dart

```diff
@@ -158,6 +343,204 @@ class SassCalculation extends Value {
+  /// Creates a `mod()` calculation with the given [dividend] and [modulus].
+  ///
+  /// Each argument must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value mod(Object dividend, Object? modulus) {
+    dividend = _simplify(dividend);
+    modulus = modulus.andThen(_simplify);
+    var args = [dividend, if (modulus != null) modulus];
+    _verifyLength(args, 2);
+    _verifyCompatibleNumbers(args);
+    if ((dividend is! SassNumber || modulus is! SassNumber) ||
+        !dividend.hasCompatibleUnits(modulus)) {
+      return SassCalculation._("mod", args);
+    }
+    return dividend.modulo(modulus);
+  }
+
+  /// Creates a `round()` calculation with the given [strategyOrNumber], [numberOrStep], and [step].
+  /// Strategy must be either nearest, up, down or to-zero.
+  ///
+  /// Number and step must be either a [SassNumber], a [SassCalculation], an
+  /// unquoted [SassString], a [CalculationOperation], or a
+  /// [CalculationInterpolation].
+  ///
+  /// This automatically simplifies the calculation, so it may return a
+  /// [SassNumber] rather than a [SassCalculation]. It throws an exception if it
+  /// can determine that the calculation will definitely produce invalid CSS.
+  ///
+  /// This may be passed fewer than two arguments, but only if one of the
+  /// arguments is an unquoted `var()` string.
+  static Value round(Object strategyOrNumber,
+      [Object? numberOrStep, Object? step]) {
+    var strategyOrNumberSimplified = _simplify(strategyOrNumber);
+
+    switch ((
+      strategyOrNumberSimplified,
+      numberOrStep.andThen(_simplify),
+      step.andThen(_simplify)
+    )) {
+      case (SassNumber number, null, null):
+        return _matchUnits(number.value.round().toDouble(), number);
+
+      case (SassNumber number, SassNumber step, null):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep('nearest', number, step);
+
+      case (SassString strategy, SassNumber number, SassNumber step)
+          when {'nearest', 'up', 'down', 'to-zero'}.contains(strategy.text):
+        _verifyCompatibleNumbers([number, step]);
+        return _roundWithStep(strategy.text, number, step);
+
+      case (
+          SassString(text: 'nearest' || 'up' || 'down' || 'to-zero'),
+          SassNumber(),
+          null
+        ):
+        throw SassScriptException("If strategy is not null, step is required.");
+
+      case (SassString strategy, SassNumber number, null):
+        return SassCalculation._("round", [strategy, number]);
```

Comment:
I don't think this is right. `round(foo, 12)` doesn't mean anything.

Reply:
There's an edge case where `foo` could contain multiple arguments right? So we should return the calculation in that case or throw an error? CC: @jathak

Reply:
You're right, I was misreading the spec here.